src/task.rs

```diff
@@ -125,7 +126,7 @@ impl<T: Send + 'static> Drop for Task<T> {
             // exit quickly: as soon as the background task has been successfully cancelled.
             spawn(async move {
                 tracing::info!(name = inner.name, "cancelling task");
-                inner.handle.cancel().await;
+                inner.handle.abort();
```

Comment:
Since tokio task cancellation is synchronous, we can get rid of this `spawn` and just call `abort` directly in the drop handler